src/Features/CSharp/Portable/CodeLens/DisplayInfoCSharpServices.cs

```diff
@@ -0,0 +1,348 @@
+                    case SyntaxKind.SingleLineDocumentationCommentTrivia:
+                        // For DocumentationCommentTrivia node, node.Parent is null. Obtain parent through ParentTrivia.Token
+                        if (node.IsStructuredTrivia)
+                        {
+                            var structuredTriviaSyntax = (StructuredTriviaSyntax) node;
+                            node = structuredTriviaSyntax.ParentTrivia.Token.Parent;
+                            continue;
+                        }
+                        return null;
+
+                    default:
+                        return node;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Gets the DisplayName for the given node.
+        /// </summary>
+        public string GetDisplayName(
+            SemanticModel semanticModel,
+            SyntaxNode node,
+            DisplayFormat displayFormat)
+        {
+            if (node == null)
+            {
+                return CSharpFeaturesResources.unknown_value;
+            }
+
+            SymbolDisplayFormat symbolDisplayFormat = DefaultFormat;
+            if (displayFormat == DisplayFormat.Short)
+            {
+                symbolDisplayFormat = ShortFormat;
+            }
+
+            string displayName;
+            string enclosingScopeString;
+
+            int lastDotBeforeSpaceIndex;
+
+            ISymbol symbol;
+
+            if (IsGlobalAttribute(node))
+            {
+                return "assembly: " + node;
+            }
+
+            // Don't discriminate between getters and setters for indexers
+            if (node.Parent != null && node.Parent.IsKind(SyntaxKind.AccessorList) &&
+                node.Parent.Parent.IsKind(SyntaxKind.IndexerDeclaration))
+            {
+                return GetDisplayName(semanticModel, node.Parent.Parent, displayFormat);
+            }
+
+            switch (node.Kind())
+            {
+                case SyntaxKind.ConstructorDeclaration:
+                    // The constructor's name will be the name of the class, not ctor like we want
+                    symbol = semanticModel.GetDeclaredSymbol(node);
+                    displayName = symbol.ToDisplayString(symbolDisplayFormat);
+                    var openParenIndex = displayName.IndexOf('(');
+                    int lastDotBeforeOpenParenIndex = displayName.LastIndexOf('.', openParenIndex, openParenIndex);
+
+                    var constructorName = symbol.IsStatic ? "cctor" : "ctor";
+
+                    displayName = displayName.Substring(0, lastDotBeforeOpenParenIndex + 1) +
+                                         constructorName +
+                                         displayName.Substring(openParenIndex);
+                    break;
+
+                case SyntaxKind.IndexerDeclaration:
+                    // The name will be "namespace.class.this[type] - we want "namespace.class[type] Indexer"
```

Comment:
i don't understand.  Why are we coming up with new formats for our symbols instead of using the standard format the language APIs produce?